src/main/java/org/jenkinsci/plugins/workflow/job/WorkflowRun.java

```diff
@@ -825,6 +646,10 @@ private String key() {
                 return TaskListener.NULL;
             }
         }
+        @Override public InputStream getLog() throws IOException {
```

Comment:
Maybe Stream is not a perfect solution for Logging. I would rather go after an event-based model similar to JUL, but it will require more changes (it's in my design btw)